Unable to run make hack-build-cli

# Checklist

* [x] I've searched the issue queue to verify this is not a duplicate bug report.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `kargo version`.
* [x] I've pasted logs, if applicable.

# Description

<!-- A clear and concise description of the bug. -->

**This isn't a bug in Kargo just in the Makefile so only relevant for developers/contributors**


When running make hack-build-cli I encounter the following error:

`docker run -it --rm -v gomodcache:/home/user/gocache -v /Users/webstradev/development/personal/kargo/:/workspaces/kargo -v /workspaces/kargo/ui/node_modules -w /workspaces/kargo kargo:dev-tools sh -c 'GOOS=darwin GOARCH=arm64 make build-cli'
CGO_ENABLED=0 go build \
        -ldflags "-w -X github.com/akuity/kargo/internal/version.version= -X github.com/akuity/kargo/internal/version.buildDate=$(date -u +'%Y-%m-%dT%H:%M:%SZ') -X github.com/akuity/kargo/internal/version.gitCommit= -X github.com/akuity/kargo/internal/version.gitTreeState=" \
        -o bin/kargo-darwin-arm64 \
        ./cmd/cli
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
make: *** [Makefile:109: build-cli] Error 1
make: *** [hack-build-cli] Error 2
webstradev@Eriks-MacBook-Pro kargo % 
`

When I add -buildvcs - false, as the error message suggests it works just fine.

Running make build-cli (without the hack) also works as intended.


# Screenshots

<!-- If applicable, add screenshots to help explain the bug. -->

# Steps to Reproduce

<!-- A list of the steps required to reproduce the issue. -->
switch to main branch, run make hack-build-cli

# Version

```
Paste the output from `kargo version` here.
```

not applicable as I can't successfully build the CLI

# Logs

```
Paste any relevant application logs here.
```
See description.

--- a/Makefile
+++ b/Makefile
@@ -170,9 +170,13 @@ codegen-ui:
 # that is pre-loaded with required tools.                                      #
 ################################################################################
 
+# Prevents issues with vcs stamping within docker containers. 
+GOFLAGS="-buildvcs=false"
+
 DOCKER_CMD := $(CONTAINER_RUNTIME) run \
 	-it \
 	--rm \
+	-e GOFLAGS=$(GOFLAGS) \
 	-v gomodcache:/home/user/gocache \
 	-v $(dir $(realpath $(firstword $(MAKEFILE_LIST)))):/workspaces/kargo \
 	-v /workspaces/kargo/ui/node_modules \

